test/broker_test/broker_test.go

```diff
@@ -36,1310 +37,1114 @@ func TestBrokers(t *testing.T) {
+					brokerID = reply.Value("id").String().Raw()
 
 					assertInvocationCount(brokerServer.CatalogEndpointRequests, 1)
+					brokerServer.ResetCallHistory()
 				})
-			})
-			Context("when an existing service offering is removed", func() {
-				var serviceOfferingID string
-
-				BeforeEach(func() {
-					catalogServiceID := gjson.Get(common.Catalog, "services.0.id").Str
-					Expect(catalogServiceID).ToNot(BeEmpty())
 
-					serviceOfferings := ctx.SMWithOAuth.GET("/v1/service_offerings").
-						Expect().
-						Status(http.StatusOK).
-						JSON().Object().Value("service_offerings").Array().Iter()
+				Context("when content type is not JSON", func() {
+					It("returns 415", func() {
+						ctx.SMWithOAuth.PATCH("/v1/service_brokers/"+brokerID).
+							WithText("text").
+							Expect().Status(http.StatusUnsupportedMediaType).
+							JSON().Object().
+							Keys().Contains("error", "description")
 
-					for _, so := range serviceOfferings {
-						sbID := so.Object().Value("broker_id").String().Raw()
-						Expect(catalogServiceID).ToNot(BeEmpty())
-
-						catalogID := so.Object().Value("catalog_id").String().Raw()
-						Expect(catalogServiceID).ToNot(BeEmpty())
-
-						if catalogID == catalogServiceID && sbID == brokerID {
-							serviceOfferingID = so.Object().Value("id").String().Raw()
-							Expect(catalogServiceID).ToNot(BeEmpty())
-							break
-						}
-					}
-					s, err := sjson.Delete(common.Catalog, "services.0")
-					Expect(err).ShouldNot(HaveOccurred())
-					brokerServer.Catalog = common.JSONToMap(s)
+						assertInvocationCount(brokerServer.CatalogEndpointRequests, 0)
+					})
 				})
 
-				It("is no longer returned by the Services and Plans API", func() {
-					plans := ctx.SMWithOAuth.GET("/v1/service_plans").
-						Expect().
-						Status(http.StatusOK).
-						JSON().Object().Value("service_plans").Array().Iter()
-
-					var planIDsForService []interface{}
-					for _, plan := range plans {
-						soID := plan.Object().Value("service_offering_id").String().Raw()
-						Expect(soID).ToNot(BeEmpty())
-						if soID == serviceOfferingID {
-							planID := plan.Object().Value("id").String().Raw()
-							Expect(soID).ToNot(BeEmpty())
-
-							planIDsForService = append(planIDsForService, planID)
-						}
-					}
-					ctx.SMWithOAuth.PATCH("/v1/service_brokers/" + brokerID).
-						WithJSON(common.Object{}).
-						Expect().
-						Status(http.StatusOK)
+				Context("when broker is missing", func() {
+					It("returns 404", func() {
+						ctx.SMWithOAuth.PATCH("/v1/service_brokers/no_such_id").
+							WithJSON(brokerServerJSON).
+							Expect().Status(http.StatusNotFound).
+							JSON().Object().
+							Keys().Contains("error", "description")
+					})
+				})
 
-					ctx.SMWithOAuth.GET("/v1/service_offerings").
-						Expect().
-						Status(http.StatusOK).
-						JSON().Path("$.service_offerings[*].id").Array().NotContains(serviceOfferingID)
+				Context("when request body is not valid JSON", func() {
```

Comment:
isn't this a generic test case for all resources?

Reply:
Post and patch can also be generic but they will come next week probably